packages/welldone-wallet/src/lib/welldone.ts

```diff
@@ -295,6 +298,36 @@ const WelldoneWallet: WalletBehaviourFactory<InjectedWallet> = async ({
       };
     },
 
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async signMessage(message: SignMessageParams): Promise<SignedMessage> {
+      if (!_state.wallet) {
+        throw new Error("Wallet is not installed");
+      }
+
+      const accounts = getAccounts();
+      const account = accounts[0];
+
+      if (!account) {
+        throw new Error("Failed to find account for signing");
+      }
```

Comment:
The way the `signMessage` should be working in wallet-selector:

A dApp should be able to trigger the `signMessage` of a wallet without first being signed-in.

```ts
const welldone = await selector.wallet("welldone-wallet");

//There should be no checks if a user is signed in or if there's an account.
const signedMessage = welldone.signMessage(params);
```
You could compare the flow of signMessage with other wallets like: Meteor, Nightly and Here Wallet:

```ts
const meteor = await selector.wallet("meteor-wallet");

// Prompts the user to sign the message without checking for the signed-in state.
const signedMessage = meteor .signMessage(params);
```

Reply:
I worry that this check will always throw unless we call `signIn` before it:

```ts
const accounts = getAccounts(); // returns [] unless we call `signIn`
const account = accounts[0];

if (!account) {
  throw new Error("Failed to find account for signing");
}
```

The `_state.account` gets assigned a value only inside the `signIn` method: https://github.com/near/wallet-selector/blob/main/packages/welldone-wallet/src/lib/welldone.ts#L245 therefore when checking for `getAccounts()` before `signIn` it will always return an `[]` empty array.

Reply:
To reproduce it, first `signOut` from Welldone in the example and open the console in Chrome dev tools and try:

```ts
const welldone = await selector.wallet("welldone-wallet");

await welldone.signMessage({
    message: "test message",
    nonce: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,24,26,27,28,29,30,31],
    recipient: "www.some-website.com",
});
```

This would allow us to add support for Welldone in this feature too: https://github.com/near/wallet-selector/pull/883

Reply:
Oh, I'm sorry. I had misunderstood our code.
You're right. Thanks to your kind explanation, I was able to fix the code :)

I modified the code to allow users and developers to call `signMessage` before `signIn`, thank you.